platform-api/src/main/java/io/datacater/core/deployment/DeploymentEndpoint.java

```diff
@@ -1,49 +1,162 @@
+import java.util.UUID;
 import javax.annotation.security.RolesAllowed;
+import javax.enterprise.context.RequestScoped;
 import javax.inject.Inject;
-import javax.persistence.criteria.CriteriaBuilder;
-import javax.persistence.criteria.CriteriaDelete;
-import javax.persistence.criteria.Root;
 import javax.ws.rs.*;
+import javax.ws.rs.core.Context;
 import javax.ws.rs.core.MediaType;
-import org.eclipse.microprofile.openapi.annotations.parameters.RequestBody;
-import org.eclipse.microprofile.openapi.annotations.security.SecurityRequirement;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.sse.OutboundSseEvent;
+import javax.ws.rs.sse.Sse;
+import javax.ws.rs.sse.SseEventSink;
 import org.hibernate.reactive.mutiny.Mutiny;
+import org.jboss.logging.Logger;
 
 @Path("/api/alpha/deployments")
 @RolesAllowed("dev")
 @Produces(MediaType.APPLICATION_JSON)
-@SecurityRequirement(name = "apiToken")
+@RequestScoped
 public class DeploymentEndpoint {
+  private static final Logger LOGGER = Logger.getLogger(DeploymentEndpoint.class);
   @Inject Mutiny.SessionFactory sf;
 
+  @Inject KubernetesClient client;
+
+  @GET
+  @Path("{deploymentName}")
+  public Uni<ObjectMeta> getStream(@PathParam("deploymentName") String deploymentName) {
+    return Uni.createFrom().item(getK8Deployment(deploymentName));
+  }
+
+  @GET
+  @Path("{deploymentName}/logs")
+  @Produces(MediaType.TEXT_PLAIN)
+  public Uni<String> getLogs(@PathParam("deploymentName") String deploymentName) {
+    return Uni.createFrom().item(getDeploymentLogs(deploymentName));
+  }
+
+  @GET
+  @Path("{deploymentName}/watchLogs")
+  @Produces(MediaType.SERVER_SENT_EVENTS)
+  public Response watchLogs(
+      @PathParam("deploymentName") String deploymentName,
+      @Context Sse sse,
+      @Context SseEventSink eventSink)
+      throws IOException {
+    LogWatch lw = watchDeploymentLogs(deploymentName).watchLog();
+    InputStream is = lw.getOutput();
+    BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(is));
+    String line;
+    while ((line = bufferedReader.readLine()) != null) {
+      final OutboundSseEvent sseEvent = sse.newEventBuilder().data(line).build();
+
+      eventSink.send(sseEvent);
+    }
+    is.close();
+    lw.close();
+    return Response.ok().build();
+  }
+
   @GET
-  public Uni<List<DeploymentEntity>> getDeployments() {
-    return sf.withSession(
-        session ->
-            session.createQuery("from DeploymentEntity", DeploymentEntity.class).getResultList());
+  public Uni<ListMeta> getDeployments() {
+    return Uni.createFrom().item(getK8Deployments());
   }
 
   @POST
-  @RequestBody
-  @Consumes(MediaType.APPLICATION_JSON)
-  public Uni<Deployment> createDeployment(Deployment deployment) throws JsonProcessingException {
-    DeploymentEntity de = new DeploymentEntity(deployment.spec());
-    return sf.withTransaction(
-        (session, transaction) -> session.merge(de).replaceWith(() -> deployment));
+  @Path("{pipelineUuid}")
```

Comment:
The `pipelineUuid` should not be part of the path but be passed as, for instance, a URI parameter, e.g., `POST /deployments/?pipeline_uuid=$pipelineUuid`